tracer/src/Datadog.Trace/IAST/Aspects/StringAspect.cs

```diff
@@ -0,0 +1,244 @@
+// <copyright file="StringAspect.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using Datadog.Trace.Iast.Dataflow;
+
+namespace Datadog.Trace.Iast.Aspects
+{
+    /// <summary> String class aspects </summary>
+    [AspectClass("mscorlib,netstandard,System.Private.CoreLib", AspectFilter.StringOptimization)]
+    [System.ComponentModel.Browsable(false)]
+    [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
+    public partial class StringAspect
```

Comment:
Talked offline about reducing or removing the array allocation in the final implementation.